src/global/global_context.cc

```diff
@@ -20,6 +20,17 @@
 #include "crimson/common/config_proxy.h"
 #endif
 
+#if defined(WITH_SEASTAR) && !defined(WITH_ALIEN)
```

Comment:
I don't understand why we're bothering to define these but make them noops.

Reply:
1. global_context.cc would be compiled to `./src/crimson/CMakeFiles/crimson-common.dir/__/global/global_context.cc.o` and
`./src/crimson/os/alienstore/CMakeFiles/crimson-alien-common.dir/__/__/__/global/global_context.cc.o`, which are `WITH_ALIEN` and `!WITH_ALIEN`. What's more, only the `WITH_ALIEN` library implements the g_ceph_context operations.

2. crimson-osd will link `WITH_ALIEN` and `!WITH_ALIEN` libraries but `g_conf_set_val` and `g_conf_rm_val` functions of the `!WITH_ALIEN` library are weak symbols. When linking, the non-weak `g_conf_set_val` and `g_conf_rm_val` of the `WITH_ALIEN` library will overwrite those of the `!WITH_ALIEN` library, that's why the `g_conf_set_val` and `g_conf_rm_val` functions do not have duplicate definition problems. Only Alienstore will alloc `g_ceph_context`, and the `WITH_ALIEN` library's `g_conf_set_val` and `g_conf_rm_val` would check the empty value of `g_ceph_context`, which ensure that !Alienstore will not have any problems.

3. Some test cases are only tested against crimson which means they would not link with the `WITH_ALIEN` library, so it is necessary to implement the empty functions of `g_conf_set_val` and `g_conf_rm_val`, otherwise undefined functions will appear.